sdk/keyvault/internal/challenge_policy.go

```diff
@@ -176,10 +176,38 @@ func (k KeyVaultChallengePolicy) getChallengeRequest(orig policy.Request) (*poli
 	req.Raw().Header.Set("Content-Length", "0")
 	req.Raw().ContentLength = 0
 
-	// copied := orig.Clone(orig.Raw().Context())
-	// copied.Raw().Body = req.Body()
+	copied := orig.Clone(orig.Raw().Context())
+	copied.Raw().Body = req.Body()
+	copied.Raw().ContentLength = 0
+	copied.Raw().Header.Set("Content-Length", "0")
+	err = copied.SetBody(NopCloser(), "application/json")
+	if err != nil {
+		return nil, err
+	}
+	copied.Raw().Header.Del("Content-Type")
+
+	return copied, err
+}
+
+// The next three methods are copied from azcore/internal/shared.go
+type nopCloser struct {
+}
+
+func (n nopCloser) Close() error {
+	return nil
+}
+
+func (n nopCloser) Read(p []byte) (int, error) {
+	return 0, nil
+}
+
+func (n nopCloser) Seek(offset int64, whence int) (int64, error) {
+	return 0, nil
+}
 
-	return req, err
+// NopCloser returns a ReadSeekCloser with a no-op close method wrapping the provided io.ReadSeeker.
+func NopCloser() io.ReadSeekCloser {
```

Comment:
This is in the `streaming` package.

https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/streaming/progress.go#L23